src/components/AppNavigator/AppNavigator.tsx

```diff
@@ -1,44 +1,114 @@
-import { useEffect } from 'react';
+import { useContext, useEffect } from 'react';
 import { Link as RouterLink, useLocation, useNavigate } from 'react-router-dom';
 import Link from '@mui/material/Link';
 import Step from '@mui/material/Step';
 import Stepper from '@mui/material/Stepper';
-import StepContent from '@mui/material/StepContent';
 import StepLabel from '@mui/material/StepLabel';
-import MenuItemType from 'types/menuItem';
+import Typography from '@mui/material/Typography';
+import FormScreens from 'constants/formScreens';
+import routes from 'constants/routes';
+import WizardFormContext from 'contexts/WizardFormContext';
+import Entries from 'types/entries';
+import { FormValuesType } from 'types/formTypes';
 
 interface Props {
-  list: MenuItemType[];
-  firstUncompletedStep: number;
+  className?: string;
 }
 
-export default function AppNavigator({ list, firstUncompletedStep }: Props) {
+export default function AppNavigator({ className }: Props) {
+  const { formValues, firstUncompletedStep } = useContext(WizardFormContext);
   const { pathname } = useLocation();
   const navigate = useNavigate();
+  // https://www.charpeni.com/blog/properly-type-object-keys-and-object-entries
+  const formValuesEntries = Object.entries(
+    formValues,
+  ) as Entries<FormValuesType>;
+  type FormValuesKeys = keyof typeof formValues;
+  type FormStepsList = (typeof formValues)[FormValuesKeys];
 
   useEffect(() => {
-    if (pathname !== list[firstUncompletedStep].url) {
-      navigate(list[firstUncompletedStep].url);
+    const formScreen = (Object.keys(routes) as (keyof typeof routes)[]).find(
```

Comment:
Check that article for `Object.entries` and apply similar approach for `Object.keys`